hera_cal/lstbin_simple.py

```diff
@@ -424,36 +469,59 @@ def lst_average(
         clip_flags[:, sc_min_N] = False
 
         flags |= clip_flags
-        data[flags] *= np.nan
+
+        if isinstance(data, np.ma.MaskedArray):
+            data.mask |= clip_flags
+        else:
+            data = np.ma.masked_array(data, mask=clip_flags)
+
         logger.info(f"Flagged a further {100*(np.sum(flags) - nflags)/flags.size:.2f}% of visibilities due to sigma clipping")
 
-    nsamples[flags] = 0
-    norm = np.sum(nsamples, axis=0)  # missing a "clip" between 1e-99 and inf here...
-    ndays_binned = np.sum(nsamples > 0, axis=0)
+        # Update the "non_inpainted" mask
+        non_inpainted = data.mask
+
+    nsamples[non_inpainted] = 0
+
+    # Here we do a check to make sure Nsamples is uniform across frequency.
+    ndiff = np.diff(nsamples, axis=2)
+    if np.any(ndiff != 0):
+        warnings.warn("Nsamples is not uniform across frequency. This will result in spectral structure.")
```

Comment:
It's a good point. I switched it so that the warning is triggered before setting non-inpainted stuff to zero, so now I think it won't be triggered by FM band. However, it now leaves open the possibility that `non_inpainted` itself might not be rectangular (i.e. it might be speckled in the waterfall). This is not the case for our current IDR, so not a problem, but we should probably check this in the long run.